### ISSUE TITLE
Invalid expression error when passing vector to CL:COUNT

### ISSUE THREAD

**Issue by pamoroso** [Interlisp/medley#1603]
Passing a vector as the second argument of the Common Lisp function `count` yields the error `invalid expression for EVAL`:

```
2/7> (count 1 #(1 2 1 3 4))
#<ARRAY T (5) @ 123,41350> is an invalid expression for EVAL.
```

The issue was discovered when evaluating the compatibility of Medley Common Lisp with ANSI Comon Lisp in issue #609, specifically by [testing the collections code](https://github.com/Interlisp/medley/issues/609#issuecomment-2002053775) of the book *Practical Common Lisp*. This is likely an [Exec issue](https://github.com/Interlisp/medley/issues/609#issuecomment-2002551818).

**Comment by masinter** [comment]
not an exec issue.

This was a Common Lisp change between CLTL1 and CLTL2.
fixed by Pr #1664       

See https://interlisp.org/clhs/Issues/iss145_w.htm

and if you're interested:
https://github.com/masinter/parcftp-cl/blob/main/cl/cleanup/old-mail/eval-other.mail

**Comment by masinter** [comment]
now fixed in master
